app/client-v2/src/stores/projectStore.ts

```diff
@@ -112,21 +107,133 @@ export const useProjectStore = defineStore("project", {
         );
       } catch (error) {
         console.error(error);
-        this.fileSamples.splice(matchedHashIndex, 1);
+        this.project.samples.splice(matchedHashIndex, 1);
       }
     },
-    // TODO: next PR to add this
+
     pollAnalysisStatus() {
-      console.log("Polling");
+      if (!this.pollingIntervalId) {
+        const intervalId = setInterval(async () => {
+          await this.getAnalysisStatus();
+        }, 2000);
+        this.pollingIntervalId = intervalId;
+      }
+    },
+
+    async getAnalysisStatus() {
+      const prevClusterAssign = this.project.status?.assign;
+      let stopPolling = false;
+      try {
+        const statusRes = await baseApi.post<ApiResponse<AnalysisStatus>>("/status", { hash: this.project.hash });
+        this.project.status = statusRes.data;
+        if (statusRes.data.assign === "finished" && prevClusterAssign !== "finished") {
+          await this.getClusterAssignResult();
+        }
+        if (
+          COMPLETE_STATUS_TYPES.includes(statusRes.data.network) &&
+          COMPLETE_STATUS_TYPES.includes(statusRes.data.microreact)
+        ) {
+          stopPolling = true;
+        }
+      } catch (error) {
+        console.error(error);
+        stopPolling = true;
+      } finally {
+        if (stopPolling) {
+          this.stopPollingStatus();
+        }
+      }
+    },
+
+    async getClusterAssignResult() {
+      try {
+        const assignClusterRes = await baseApi.post<ApiResponse<AssignCluster>>("/assignResult", {
+          projectHash: this.project.hash
+        });
+
+        Object.values(assignClusterRes.data).forEach((clusterInfo: ClusterInfo) => {
+          const matchedHashIndex = this.project.samples.findIndex(
+            (sample: ProjectSample) => clusterInfo.hash === sample.hash
+          );
+          if (matchedHashIndex !== -1) {
+            this.project.samples[matchedHashIndex].cluster = clusterInfo.cluster;
+          }
+        });
+      } catch (error) {
+        console.error(error);
+      }
+    },
+
+    stopPollingStatus() {
+      if (this.pollingIntervalId) {
+        clearInterval(this.pollingIntervalId);
+        this.pollingIntervalId = null;
+      }
     },
     // TODO: update to remove from api as well
     removeUploadedFile(index: number) {
-      this.fileSamples.splice(index, 1);
+      // this.fileSamples.splice(index, 1);
     },
-    // TODO: next PR to add this
+
     async runAnalysis() {
-      console.log("Run");
-      this.isRun = true;
+      const body = this.buildRunAnalysisPostBody();
+      try {
+        await baseApi.post("/poppunk", body);
+        this.project.hash = body.projectHash;
+      } catch (error) {
+        console.error("Error running analysis", error);
```

Comment:
I think I've already asked this, but there is a ticket for putting in error handling generally?

Reply:
yup